pkg/controller/persistentvolume/persistentvolume_claim_binder_controller.go

```diff
@@ -282,6 +282,15 @@ func syncClaim(volumeIndex *persistentVolumeOrderedIndex, binderClient binderCli
 			return fmt.Errorf("A volume match does not exist for persistent claim: %s", claim.Name)
 		}
 
+		// create a reference to the claim and assign it to the volume being bound.
+		// the volume is a pointer and assigning the reference fixes a race condition where another
+		// claim might match this volume but before the claimRef is persistent in the next case statement
+		claimRef, err := api.GetReference(claim)
+		if err != nil {
+			return fmt.Errorf("Unexpected error getting claim reference: %v\n", err)
+		}
+		volume.Spec.ClaimRef = claimRef
```

Comment:
People should not edit the state of an object in a cache.Store/cache.Index - period.

Is there no other alternative?